backend/drive/drive.go

```diff
@@ -684,8 +691,16 @@ func (f *Fs) list(ctx context.Context, dirIDs []string, title string, directorie
 		if parentsQuery.Len() > 1 {
 			_, _ = parentsQuery.WriteString(" or ")
 		}
-		if f.opt.SharedWithMe && dirID == f.rootFolderID {
-			_, _ = parentsQuery.WriteString("sharedWithMe=true")
+		if dirID == f.rootFolderID {
+			if f.opt.SharedWithMe {
+				_, _ = parentsQuery.WriteString("sharedWithMe=true")
+			}
+			if f.opt.StarredOnly {
+				if f.opt.SharedWithMe {
+					_, _ = parentsQuery.WriteString(" and ")
```

Comment:
I'm not sure this `and` is correct without brackets - what do you think?

Reply:
It seems to work for me as is....but I'll defer to whatever seems best.

My bigger annoyance is that "starred" and "sharedwithme" seem not to play very well together.  "sharedwithme" seems to retain parent/child folder structures, but starred does not.  I guess it is more of how I wanted to use it.  For example, I had hoped that by "staring" a specific file, it would retain its parent/child folder relationships.  But I admit that would not follow how Google has it setup through its UI whereby when you star an item it only shows that item instead of its full file path.

Reply:
If you want to show the starred files in their correct place then you want something like this

```go
		if f.opt.SharedWithMe && dirID == f.rootFolderID {
			_, _ = parentsQuery.WriteString("sharedWithMe=true")
		} else {
			_, _ = fmt.Fprintf(parentsQuery, "'%s' in parents", dirID)
		}
		if f.opt.StarredOnly {
			if parentsQuery.Len() > 1 {
				_, _ = parentsQuery.WriteString(" and ")
			}
			_, _ = parentsQuery.WriteString("(mimeType='" + driveFolderType + "' or starred=true)")
		}
```

That will make --shared-with-me incompatible with --starred-only though as it searches in given directories only.

If you wanted starred to work with sharedwith me then you'd want something like (untested)

```go
		if f.opt.SharedWithMe && dirID == f.rootFolderID {
			_, _ = parentsQuery.WriteString("sharedWithMe=true")
			if f.opt.StarredOnly {
				if parentsQuery.Len() > 1 {
					_, _ = parentsQuery.WriteString(" and ")
				}
				_, _ = parentsQuery.WriteString("starred=true")
			}
		} else {
			_, _ = fmt.Fprintf(parentsQuery, "'%s' in parents", dirID)
			if f.opt.StarredOnly {
				if parentsQuery.Len() > 1 {
					_, _ = parentsQuery.WriteString(" and ")
				}
				_, _ = parentsQuery.WriteString("(mimeType='" + driveFolderType + "' or starred=true)")
			}
		}
```